### PR TITLE
Creates CircleCI config for repo

### PR DESCRIPTION
**Pull request by conorsch**
Runs:

  go fmt
  go vet
  go build

And then a "help" on the built binary itself, as a sanity check.
Includes small tweaks required to get CI passing, including some type
changes and a "go fmt" on the new sddir module.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -239,7 +239,7 @@ func scan(cmd *cli.Cmd) {
 			f, err := os.Create(*outputFile)
 			defer f.Close()
 			if err != nil {
-				log.Fatalf("Could not open output file %s: %s", outputFile, err)
+				log.Fatalf("Could not open output file %s: %s", *outputFile, err)
```
Comment (conorsch) [inline main.go:242]: These changes were recommended by `go vet`.

File: sddir.go [hunk 2]
```diff
@@ -2,34 +2,34 @@ package main
 
 import (
 	"encoding/json"
+	log "github.com/sirupsen/logrus"
```
Comment (conorsch) [inline sddir.go:5]: All changes to `sddir.go` were simply `go fmt`.

File: main.go (hunk 1 above)
Comment (rmol) [reply main.go:242]: it was (unsurprisingly) correct.

File: sddir.go (hunk 2 above)
Comment (rmol) [reply sddir.go:5]: cool. thought I had that running on save, but clearly I need to go spend two hours on my Emacs config.

**Review by rmol — APPROVED** [review]
Looks good.

**Pull request merged**